### ISSUE TITLE
device class not set if there is no metadata device defind

### ISSUE THREAD

**Issue by jshen28** [rook/rook#6818]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

I define a cephcluster like follows, with deviceClass key set,

```yaml
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  creationTimestamp: "2020-12-11T02:32:57Z"
  finalizers:
  - cephcluster.ceph.rook.io
  generation: 2
  name: rook-ceph
  namespace: rook-ceph
  resourceVersion: "58144742"
  selfLink: /apis/ceph.rook.io/v1/namespaces/rook-ceph/cephclusters/rook-ceph
  uid: 2e46a23d-3b59-11eb-a758-52540061a481
spec:
  cephVersion:
    image: registry-jinan-lab.inspurcloud.cn/library/ceph/ceph:14.2.15-li1
  cleanupPolicy:
    sanitizeDisks: {}
  crashCollector:
    disable: false
  dashboard:
    enabled: true
  dataDirHostPath: /var/lib/rook
  disruptionManagement: {}
  external:
    enable: false
  healthCheck:
    daemonHealth:
      mon: {}
      osd: {}
      status: {}
    livenessProbe:
      mon:
        disabled: true
      osd:
        disabled: true
  mgr: {}
  mon:
    count: 3
  monitoring: {}
  network:
    hostNetwork: true
    provider: ""
    selectors: null
  placement:
    mgr:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/mgr
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
    mon:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/mon
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
    osd:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/osd
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
  removeOSDsIfOutAndSafeToRemove: false
  resources:
    mgr:
      limits:
        cpu: "2"
        memory: 4Gi
      requests:
        cpu: 300m
        memory: 512Mi
    mon:
      limits:
        cpu: "2"
        memory: 24Gi
      requests:
        cpu: 300m
        memory: 512Mi
    osd:
      limits:
        cpu: "2"
        memory: 8Gi
      requests:
        cpu: 300m
        memory: 512Mi
  security:
    kms: {}
  storage:
    config:
      databaseSizeMB: ""
      metadataDevice: ""
      storeType: bluestore
      walSizeMB: ""
    nodes:
    - config: null
      devices:
      - config:
          deviceClass: sjt-test
        name: vdc
      - config: null
        name: vdd
      name: rookceph01
      resources: {}
    - config: null
      devices:
      - config: null
        name: vdc
      - config: null
        name: vdd
      name: rookceph02
      resources: {}
    - config: null
      devices:
      - config: null
        name: vdc
      - config: null
        name: vdd
      name: rookceph03
      resources: {}
    storageClassDeviceSets: null
    useAllDevices: false
```

I saw config in env variables in prepare pod:

```yaml
    - name: ROOK_DATA_DEVICES
      value: '[{"id":"vdc","storeConfig":{"osdsPerDevice":1,"deviceClass":"sjt-test"}},{"id":"vdd","storeConfig":{"osdsPerDevice":1}}]'
```

But after cluster status is ok,

```console
root@rookceph01:~/sjt/src# kubectl get cephclusters.ceph.rook.io -n rook-ceph
NAME        DATADIRHOSTPATH   MONCOUNT   AGE    PHASE   MESSAGE                        HEALTH
rook-ceph   /var/lib/rook     3          102m   Ready   Cluster created successfully   HEALTH_OK
```

device class is not set correctly

```console
root@rookceph01:~/sjt/src# kubectl -n rook-ceph exec -it rook-ceph-tools-67f5bd8f6c-8h7gm -- ceph osd tree
ID CLASS WEIGHT  TYPE NAME           STATUS REWEIGHT PRI-AFF 
-1       0.58612 root default                                
-3       0.19537     host rookceph01                         
 0   hdd 0.09769         osd.0           up  1.00000 1.00000 
 3   hdd 0.09769         osd.3           up  1.00000 1.00000 
-5       0.19537     host rookceph02                         
 1   hdd 0.09769         osd.1           up  1.00000 1.00000 
 4   hdd 0.09769         osd.4           up  1.00000 1.00000 
-7       0.19537     host rookceph03                         
 2   hdd 0.09769         osd.2           up  1.00000 1.00000 
 5   hdd 0.09769         osd.5           up  1.00000 1.00000 
```

**Expected behavior:**

device class flag set

**How to reproduce it (minimal and precise):**

deploy a cephcluster:

1. without any metadata device
2. set device class flag

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.3 LTS
* Kernel (e.g. `uname -a`):  5.0.0-29-generic

**Comment by jshen28** [comment]
#https://github.com/rook/rook/pull/6817

**Comment by travisn** [comment]
This is covered by #6814 and will be fixed in #6815. Good timing. :)
